### PR TITLE
Cliffd/file refactor

### PR DESCRIPTION
**Pull request by cliff-dellamano**
# `py-6x-corrections` Initial Release

## What?
Looks like it's time, folks. This PR is a full refactor of the library, so there's a lot to go through here. Big ticket items:
* The helper functions that were originally grouped together with the `correct_6x_images` scripts have been separated out into modules that make more sense according to their use. Also, most of the actual business logic was refactored into functions as well.
* Reflectance calibration via calibration panels has been integrated into the script (thanks to @dzermas)
* Many use-cases requiring the parsing of EXIF/XMP metadata have been added to `py-image-metadata-parser` as first-class citizens.

## Why?
* This will mark the first "release" of the library that is suitable for public use. There are still things that can be done to improve the user experience (`click` CLI bindings) as well as better practices from a developer standpoint (test suite, documentation, etc.), but these can be done post-release. 

## PR Checklist
- [x] Merged latest master
- [x] Updated version number
- [x] All private git packages are at their newest version in *environment.yml*

### CODE REVIEW

File: correct6x/detect_panel.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    """
+    top_left: Tuple[int, int]
+    bottom_right: Tuple[int, int]
+
+    def bounds(self):
+        """Return a sequence of slice objects that allow for use of the bounding box in NumPy indexing expressions."""
+        return slice(self.top_left[1], self.bottom_right[1]), slice(self.top_left[0], self.bottom_right[0])
+
+
+def convert_to_type(image, desired_type=np.uint8):
```
Comment (dzermas) [inline correct6x/detect_panel.py:33]: This should probably be called ` convert_12bit_to_type`

**Review by dzermas — CHANGES_REQUESTED** [review]
LGTM. Just consider changing the name of the `convert_to_type` function.

**Review by ZachMarston — APPROVED** [review]
LGTM, pending Dimitris' requested change.

**Review by dzermas — APPROVED** [review]
LGTM

**Pull request merged**